pkg/agent/baker.go

```diff
@@ -48,12 +54,22 @@ func (t *TemplateGenerator) getLinuxNodeCustomDataJSONObject(config *NodeBootstr
 		panic(e)
 	}
 
-	return fmt.Sprintf("{\"customData\": \"[base64(concat('%s'))]\"}", str)
+	return str
 }
 
-// GetWindowsNodeCustomDataJSONObject returns Windows customData JSON object in the form
-// { "customData": "[base64(concat(<customData string>))]" }
-func (t *TemplateGenerator) getWindowsNodeCustomDataJSONObject(config *NodeBootstrappingConfiguration) string {
+// GetLinuxNodeCustomDataJSONObject returns Linux customData JSON object in the form
+// { "customData": "[base64(<customData string>)]" }
+func (t *TemplateGenerator) getLinuxNodeCustomDataJSONObjectARM(config *NodeBootstrappingConfiguration) string {
+	return fmt.Sprintf("{\"customData\": \"[base64('%s')]\"}", t.getLinuxNodeCustomDataJSONObjectCommon(config))
+}
+
+func (t *TemplateGenerator) getLinuxNodeCustomDataJSONObjectNonARM(config *NodeBootstrappingConfiguration) string {
+	customData := t.getLinuxNodeCustomDataJSONObjectCommon(config)
+	base64CustomData := base64.StdEncoding.EncodeToString([]byte(customData))
+	return fmt.Sprintf("{\"customData\": \"%s\"}", base64CustomData)
+}
+
+func (t *TemplateGenerator) getWindowsNodeCustomDataJSONObjectCommon(config *NodeBootstrappingConfiguration) string {
```

Comment:
not sure if we need to touch windows in this PR - windows cse/custom data is not yet migrated to AgentBaker, what we have here is basically garbage data.

Reply:
I have removed Windows related change.